frontend/src/app/offers/create-offer/create-offer.component.ts

```diff
@@ -0,0 +1,192 @@
+import 'rxjs/add/operator/filter';
+import 'rxjs/add/operator/do';
+import { Component, OnDestroy, OnInit, ViewChild } from '@angular/core';
+import { AppOffer, ApiOffer, BaseOffer } from 'app/homepage-offer/offers.model';
+import { AuthService } from 'app/auth.service';
+import { User } from 'app/user';
+import { Observable } from 'rxjs/Observable';
+import { OffersService } from 'app/homepage-offer/offers.service';
+import { ActivatedRoute, Params, Router } from '@angular/router';
+import { environment } from 'environments/environment';
+import { Subscription } from 'rxjs/Subscription';
+import { FileReaderEvent, FileReaderEventTarget } from '../../models';
+import { NgForm, FormGroup } from '@angular/forms';
+import { FormBuilder } from '@angular/forms';
+import { Validators } from '@angular/forms';
+import { Image } from 'app/homepage-offer/image.model';
+import { Response } from '@angular/http';
+@Component({
+  selector: 'volontulo-create-offer',
+  templateUrl: './create-offer.component.html',
+  styleUrls: ['./create-offer.component.scss']
+})
+export class CreateOfferComponent implements OnInit, OnDestroy {
+  public djangoRoot = environment.djangoRoot;
+  public hasOrganization = false;
+  public inEditMode = false;
+  public isAdmin = false;
+  public offer: AppOffer = new AppOffer;
+  public reader = new FileReader();
+  public user: User;
+  public userSubscription: Subscription;
+  public form: FormGroup;
+  public error;
+
+  constructor(
+    private authService: AuthService,
+    private offersService: OffersService,
+    private activatedRoute: ActivatedRoute,
+    private fb: FormBuilder,
+    private router: Router,
+  ) {}
+
+  ngOnInit() {
+    this.form = this.fb.group({
+      title: ['', Validators.required],
+      location: ['', Validators.required],
+      organization: ['', Validators.required],
+      startedAt: ['', Validators.required],
+      actionOngoing: [false],
+      finishedAt: ['', Validators.required],
+      constantCoop: [false],
+      recruitmentStartDate: [null],
+      recruitmentEndDate: [null],
+      volunteersLimit: [],
+      reserveRecruitmentStartDate: [null],
+      reserveRecruitmentEndDate: [null],
+      reserveVolunteersLimit: [],
+      description: ['', Validators.required],
+      timeCommitment: ['', Validators.required],
+      benefits: ['', Validators.required],
+      requirements: [''],
+      image: [],
+    })
+
+      this.userSubscription = this.authService.user$
+      .subscribe(
+        response => {
+          this.user = response;
```

Comment:
This function introduces a bug.

user$ is an observable on BehaviourSubject. This BehaviourSubject is initialized in authService with "null". When you subscribe to BehaviourSubject/Observable based on BehaviourSubject it returns the last observed value. When no user has logged in or the request for currentUser haven't finished yet, it returns 'null'. That what is happening when you refresh the page on this component. Angular starts from the ground up and it receives 'null' here. You then try to read some properties from `this.user` object, but it is not an object, since it is 'null' and we get a 'ReferenceError' exception, since properties are undefined. However, Angular catches the error (I think it shouldn't, it should log it to console) so we don't see it in the console, but the page does not load properly.

To write it in a more proper way:
```
.subscribe(user => {
   if(user) {
     this.user = user;
     this.isAdmin = user['isAdministrator'];
     this.hasOrganization = user['organizations'].length > 0;
   }
})
```
Also in a .html file we should use user?.organizations instead of user.organizations to account for the possible 'null' value.


I am also not sure if we should use `user['isAdministrator']` vs `user.isAdministrator`. From the JavaScript perspective it is correct, however production compiler might minify it differently depending on how it is written and it would result in some errors. Let's leave it for now, we will see how it works when it gets to production environment. :)